crates/storage/db/src/tables/codecs/compact.rs

```diff
@@ -121,3 +121,85 @@ macro_rules! add_wrapper_struct {
 add_wrapper_struct!((U256, CompactU256));
 add_wrapper_struct!((u64, CompactU64));
 add_wrapper_struct!((ClientVersion, CompactClientVersion));
+
+#[cfg(test)]
+mod tests {
+    use crate::{
+        codecs::{
+            compact::{CompactClientVersion, CompactU64},
+            CompactU256,
+        },
+        models::{StoredBlockBodyIndices, StoredBlockOmmers, StoredBlockWithdrawals},
+    };
+    use reth_primitives::{
+        stage::{
+            AccountHashingCheckpoint, CheckpointBlockRange, EntitiesCheckpoint,
+            ExecutionCheckpoint, HeadersCheckpoint, IndexHistoryCheckpoint, StageCheckpoint,
+            StageUnitCheckpoint, StorageHashingCheckpoint,
+        },
+        Account, Header, PruneCheckpoint, PruneMode, PruneSegment, Receipt, ReceiptWithBloom,
+        SealedHeader, TxEip1559, TxEip2930, TxEip4844, TxLegacy, Withdrawal, Withdrawals,
+    };
+
+    #[test]
+    fn test_ensure_backwards_compatibility() {
```

Comment:
Just for posterity let's add a comment on the test explaining why it's doing these checks